### PR TITLE
Rename base to catalog

### PR DESCRIPTION
**Pull request by ThunderDeliverer**
# Description

As we renamed `Base` to `Catalog` our contracts had to be updated.

# Actions

- Renamed smart contracts
- Updated references to `Base` with `Catalog`
- Updated NatSpec comments to reflect the changes
- Updated tests to reference `catalog` rather than `base`

# Checklist

- [x] Verified code additions
- [x] Updated NatSpec comments (if applicable)
- [x] Regenerated docs
- [x] Ran prettier
- [x] Added tests to fully cover the changes

### CODE REVIEW

File: contracts/RMRK/equippable/IRMRKEquippable.sol [hunk 1]
```diff
@@ -133,13 +133,13 @@ interface IRMRKEquippable is IRMRKMultiAsset {
      *      childEquippableAddress
      *  ]
      * @param tokenId ID of the token for which we are retrieving the equipped object
-     * @param targetBaseAddress Address of the `Base` associated with the `Slot` part of the token
+     * @param targetcatalogAddress Address of the `Catalog` associated with the `Slot` part of the token
```
Comment (steven2308) [inline contracts/RMRK/equippable/IRMRKEquippable.sol:136]:
```suggestion
     * @param targetCatalogAddress Address of the `Catalog` associated with the `Slot` part of the token
```

File: contracts/RMRK/equippable/IRMRKEquippable.sol [hunk 2]
```diff
@@ -133,13 +133,13 @@ interface IRMRKEquippable is IRMRKMultiAsset {
...
      * @param slotPartId ID of the `Slot` part that we are checking for equipped objects
      * @return struct The `Equipment` struct containing data about the equipped object
      */
     function getEquipment(
         uint256 tokenId,
-        address targetBaseAddress,
+        address targetcatalogAddress,
```
Comment (steven2308) [inline contracts/RMRK/equippable/IRMRKEquippable.sol:142]:
```suggestion
        address targetCatalogAddress,
```

File: contracts/RMRK/equippable/RMRKEquippable.sol [hunk 3]
```diff
@@ -450,15 +450,15 @@ contract RMRKEquippable is
         uint64 assetId,
         uint64 slotPartId
     ) internal virtual {
-        address targetBaseAddress = _baseAddresses[assetId];
-        Equipment memory equipment = _equipments[tokenId][targetBaseAddress][
+        address targetcatalogAddress = _catalogAddresses[assetId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKEquippable.sol:453]:
```suggestion
        address targetCatalogAddress = _catalogAddresses[assetId];
```

File: contracts/RMRK/equippable/RMRKEquippable.sol [hunk 4]
```diff
@@ -450,15 +450,15 @@ contract RMRKEquippable is
...
+        Equipment memory equipment = _equipments[tokenId][targetcatalogAddress][
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKEquippable.sol:454]:
```suggestion
        Equipment memory equipment = _equipments[tokenId][targetCatalogAddress][
```

File: contracts/RMRK/equippable/RMRKEquippable.sol [hunk 5]
```diff
@@ -450,15 +450,15 @@ contract RMRKEquippable is
...
             slotPartId
         ];
         if (equipment.childEquippableAddress == address(0))
             revert RMRKNotEquipped();
         _beforeUnequip(tokenId, assetId, slotPartId);
 
-        delete _equipments[tokenId][targetBaseAddress][slotPartId];
+        delete _equipments[tokenId][targetcatalogAddress][slotPartId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKEquippable.sol:461]:
```suggestion
        delete _equipments[tokenId][targetCatalogAddress][slotPartId];
```

File: contracts/RMRK/equippable/RMRKEquippable.sol [hunk 6]
```diff
@@ -558,10 +558,10 @@ contract RMRKEquippable is
      */
     function getEquipment(
         uint256 tokenId,
-        address targetBaseAddress,
+        address targetcatalogAddress,
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKEquippable.sol:561]:
```suggestion
        address targetCatalogAddress,
```

File: contracts/RMRK/equippable/RMRKEquippable.sol [hunk 7]
```diff
@@ -558,10 +558,10 @@ contract RMRKEquippable is
...
         uint64 slotPartId
     ) public view virtual returns (Equipment memory) {
-        return _equipments[tokenId][targetBaseAddress][slotPartId];
+        return _equipments[tokenId][targetcatalogAddress][slotPartId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKEquippable.sol:564]:
```suggestion
        return _equipments[tokenId][targetCatalogAddress][slotPartId];
```

File: contracts/RMRK/equippable/RMRKExternalEquip.sol [hunk 8]
```diff
@@ -422,15 +422,15 @@ contract RMRKExternalEquip is
...
+        address targetcatalogAddress = _catalogAddresses[assetId];
+        Equipment memory equipment = _equipments[tokenId][targetcatalogAddress][
             slotPartId
         ];
         if (equipment.childEquippableAddress == address(0))
             revert RMRKNotEquipped();
 
         _beforeUnequip(tokenId, assetId, slotPartId);
-        delete _equipments[tokenId][targetBaseAddress][slotPartId];
+        delete _equipments[tokenId][targetcatalogAddress][slotPartId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKExternalEquip.sol:433]:
```suggestion
        delete _equipments[tokenId][targetCatalogAddress][slotPartId];
```

File: contracts/RMRK/equippable/RMRKExternalEquip.sol [hunk 9]
```diff
@@ -422,15 +422,15 @@ contract RMRKExternalEquip is
         uint64 assetId,
         uint64 slotPartId
     ) internal virtual {
-        address targetBaseAddress = _baseAddresses[assetId];
-        Equipment memory equipment = _equipments[tokenId][targetBaseAddress][
+        address targetcatalogAddress = _catalogAddresses[assetId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKExternalEquip.sol:425]:
```suggestion
        address targetCatalogAddress = _catalogAddresses[assetId];
```

File: contracts/RMRK/equippable/RMRKExternalEquip.sol [hunk 10]
```diff
@@ -422,15 +422,15 @@ contract RMRKExternalEquip is
...
+        Equipment memory equipment = _equipments[tokenId][targetcatalogAddress][
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKExternalEquip.sol:426]:
```suggestion
        Equipment memory equipment = _equipments[tokenId][targetCatalogAddress][
```

File: contracts/RMRK/equippable/RMRKExternalEquip.sol [hunk 11]
```diff
@@ -566,10 +566,10 @@ contract RMRKExternalEquip is
      */
     function getEquipment(
         uint256 tokenId,
-        address targetBaseAddress,
+        address targetcatalogAddress,
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKExternalEquip.sol:569]:
```suggestion
        address targetCatalogAddress,
```

File: contracts/RMRK/equippable/RMRKExternalEquip.sol [hunk 12]
```diff
@@ -566,10 +566,10 @@ contract RMRKExternalEquip is
...
         uint64 slotPartId
     ) public view returns (Equipment memory) {
-        return _equipments[tokenId][targetBaseAddress][slotPartId];
+        return _equipments[tokenId][targetcatalogAddress][slotPartId];
```
Comment (steven2308) [inline contracts/RMRK/equippable/RMRKExternalEquip.sol:572]:
```suggestion
        return _equipments[tokenId][targetCatalogAddress][slotPartId];
```

**Review by steven2308 — CHANGES_REQUESTED** [review]
Minor camel case fix.

**Review by stevyhacker — APPROVED** [review]

**Review by steven2308 — APPROVED** [review]

**Pull request merged**